### PR TITLE
Added RenderGeometry handling and RenderMesh method from RenderGeometry

### LINKED ISSUE #105: Querying RenderMesh of a RenderGeometry causes a Stack Overflow.
#### Description:
The `RenderMesh(obj)` method in Graphics_Engine is called recursively as a result of the `IRenderMesh` method not handling `RenderGeometry`. 

#### Steps to reproduce:
Plug a `RenderGeometry` into `Query.RenderMesh`

#### Expected behaviour:
`RenderMesh` of `RenderGeometry` returned with color and geometry as appropriate.

### PR DESCRIPTION
**Pull request by enarhi**
### Issues addressed by this PR

Closes #105 

Added handling and method for getting RenderMesh from RenderGeometry.

### Test files
[test file](https://burohappold.sharepoint.com/sites/BHoM/02_Current/Forms/AllItems.aspx?newTargetListUrl=%2Fsites%2FBHoM%2F02%5FCurrent&viewpath=%2Fsites%2FBHoM%2F02%5FCurrent%2FForms%2FAllItems%2Easpx&id=%2Fsites%2FBHoM%2F02%5FCurrent%2F12%5FScripts%2F02%5FPull%20Request%2FBHoM%2FTriangleNet%5FToolkit%2F%23107%2DRenderMeshFromRenderGeometry&viewid=a5a93cba%2Dfcca%2D46cc%2Da31a%2D80b8d5cbfd7b)

### CODE REVIEW

**Comment by enarhi** [comment]
@BHoMBot check compliance

File: TriangleNet_Engine/Compute/Representation/RenderMesh/Other/RenderGeometry.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    {
+        /***************************************************/
+        /**** Public Methods - Graphics                 ****/
+        /***************************************************/
+
+        [Description("Returns the RenderMesh representation of an input RenderGeometry.")]
+        [Input("renderGeometry", "The RenderGeometry to get a RenderMesh from.")]
+        [Input("renderMeshOptions", "Options that regulate both the calculation of the Geometrical Representation and how it should be meshed.")]
+        [Output("A RenderMesh, which is a geometrical mesh that can potentially have additional attributes like Colours.")]
+
```
Comment (FraserGreenroyd) [inline TriangleNet_Engine/Compute/Representation/RenderMesh/Other/RenderGeometry.cs]:
```suggestion
```

File: TriangleNet_Engine/Compute/Representation/RenderMesh/Other/RenderGeometry.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        public static BH.oM.Graphics.RenderMesh RenderMesh(this RenderGeometry renderGeometry, RenderMeshOptions renderMeshOptions = null)
```
Comment (FraserGreenroyd) [inline TriangleNet_Engine/Compute/Representation/RenderMesh/Other/RenderGeometry.cs]: I think file name needs to be updated looking at the code?

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

**Comment by enarhi** [comment]
@BHoMBot check compliance

**Review by FraserGreenroyd — APPROVED** [review]
I'm happy from a DevOps perspective after code review. Will let @alelom do a final functionality review for a final bit of testing 😄

**Review by alelom — APPROVED** [review]
Approving functionality from test script, also code looks fine.

**Comment by enarhi** [comment]
@BHoMBot check required

**Comment by FraserGreenroyd** [comment]
@BHoMBot check required

**Comment by FraserGreenroyd** [comment]
@BHoMBot check unit-tests
@BHoMBot check ready-to-merge

**Pull request merged**